### PR TITLE
.CRTChanges

### PR DESCRIPTION
**Pull request by RaquelCardona**
Algorithm & Discussion Protocols Added

### CODE REVIEW

**Comment by nathanielgranor** [comment]
@Kenney-Chan, how do you want to handle the review process for Raquel's incremental additions?

**Comment by hollasch** [comment]
@nathanielgranor -- I believe Kenney is using the @kenneychan account.

**Comment by kenneychan** [comment]
@nathanielgranor, @hollasch, @RaquelCardona Raquel should assign me and you as reviewers.  Either one of us can review and approve the CRT changes.  

@Raquel there is typo algoritm should be algorithm.

File: lesson_04.md [hunk 1]
```diff
@@ -54,8 +54,8 @@ Students will be able to...
...
 * For students that are hesitant or unwilling to engage, point out some simple, useful features (like basic drawing and the ![](say.png) block) to get them started.  Remind them that complexity or "coolness" are not important for this lab.
-* If you wish, you can add a second day to this lesson for students to share out their programs.  Work with your classroom teacher and/or school IT staff to determine the best way for students to be able to present.
-    * One simple option is to have all students share their projects with you, and then show each one in turn on the teacher computer/projector.
+* If you wish, you can add a second day to this lesson for students to share out their programs on a volunteer basis.  Work with your classroom teacher and/or school IT staff to determine the best way for students to be able to present.
+    * One simple option is to have all students share their projects with you, and then show each one in turn on the teacher computer/projector of those students that volunteer.
```
Comment (hollasch) [inline lesson_04.md]: "... on the teacher computer/projector of those students that volunteer". This sentence doesn't parse. Consider "... is to have volunteering students share ...".

File: lesson_12.md [hunk 2]
```diff
@@ -39,12 +40,15 @@ Students will be able to...
...
         -   Students should turn in answers to all questions and SNAP programs for the final problems
 
 3.  Debrief
     -   Go through each question or prompt in parts 1 and 2 and ask students to share their answer
-        -   Cold calling works well for this lab due to the large number of questions
+        -   Cold calling alternative works well for this lab due to the large number of questions
+            -  **Cold calling alternative:** Write each student’s name on an index card and shuffle the cards. Next, ask a few “review” questions to set the stage that ask for application (or creativity, or evaluation) and have no single “right answer.” Give students a little time to formulate an answer- 30-60 seconds. You can have them talk in pairs or groups for a few minutes as well if you prefer. It’s only after students have been thinking that you take the top card to see who answers. Sometimes you can say something like, “If you don’t know, make something up like you would on a mid-term, and then we’ll all help you develop better answers.” The student takes their best shot, knowing that others will help if necessary.
```
Comment (hollasch) [inline lesson_12.md]: Recommend "Give students 30 to 60 seconds to formulate...".

File: lesson_12.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ At that point you have lots of options:  You can ask a follow-up to the same stu
 ## Accommodations/Differentiation
 
 -   Colorblind students may not be able to identify the block colors, but can still recognize the organization of categories. Be sensitive to this, but no modifications are likely required.
--   Students that are struggling with the lab can be paired up and/or receive individual instructor attention to help them through the activity.
+-   Students that are struggling with the lab can be paired up and/or receive individual instructor attention to help them through the activity. Use you could also use ["Helping Trios]
```
Comment (hollasch) [inline lesson_12.md]: "... could also use the _Helping Trios_ worksheet."

File: lesson_13.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ Students will be able to...
 
 ## Materials/Preparation
 
--   [Do Now 1.3: Drawing a Triangle](do_now_13.md)  
+-   [Do Now 1.3: Drawing a Triangle](do_now_13.md) You can ask students to gives instructions in plain language, pseudo code or with images
```
Comment (hollasch) [inline lesson_13.md]: "... students to give instructions ...". Also "plain language, pseudo code, or images".

**Comment by RaquelCardona** [comment]
@hollasch I was having trouble submitting multiple pull requests. All changes I made automatically added to the initial pull request.  To solve this, I believe I should create a branch and submit pull requests from the branch. Is this correct?
Kenney early on mentioned I should not create a branch but maybe I misunderstood.

I will start the new procedure, incorporating your suggested best practices, starting with unit 2.

**Comment by hollasch** [comment]
@RaquelCardona — your workflow is mostly ok (fork + PR). An alternative approach that I find much easier is to create a feature branch from `master` in the primary repo. In other words, clone `introduction-to-computer-science`, create a branch named something like  `CRT-update`, and when you're done working in that branch, issue a pull request to merge that branch back into `master`.

The advantage of this approach is that I a reviewer can checkout that branch, and do several things:

1. Review binaries, like DOCX files,
2. Create additional commits with changes for _you_ to review. Sometimes this is a lot easier than writing up a bunch of tiny pieces of feedback and having you do them.

Regarding the "many commits" issue, you only need to commit when you have a set of changes that make sense together. Having these chain onto the existing pull request is fine, as you can create a _new_ branch on top of the branch currently under review. However, the issue I'm commenting on is creating separate commits for every sentence. That creates fairly noisy commit chains, particularly cases where a commit is a fragment of an unfinished whole.

Without going too deep, at some point you might find it interesting to read up on Git rebasing. Prior to a pull request, this allows to you squash some commits together, abandon some, reorder changes, and even edit commits, until you have a nice series of changes that are ready to present. If you're not familiar with rebasing, no worries, but let me know if you're interested and have questions.

**Pull request merged**